src/block.js

```diff
@@ -165,6 +193,14 @@ class Block {
     return this.merkleRoot.compare(actualMerkleRoot) === 0
   }
 
+  checkWitnessCommit () {
+    if (!this.transactions) return false
+    if (!this.isWitnessBlock()) return false
```

Comment:
I wonder if we should return true here...

Since the no transactions check is saying "Having no transactions is an invalid state therefore false"

But a non-witness-block not having a witness commit to check is the expected valid state.

So if I run checkWitnessCommit on a block with no witness, maybe we should return true???

Reply:
If we don't do this, every consumer of this API will need to dance around checking "if isWitnessBlock, then checkWitnessCommit, else don't" etc...

Maybe we want them to do this?

Reply:
I don't think `true` is acceptable.
This is a misunderstanding of the situation no matter how you look at it.
The result is nothing was done,  so maybe `null`,  but honestly,  `Error` seems appropriate.

Like calling `.privateKey` on `ECPair` with only `.publicKey`.

Reply:
Yeah.

Maybe I should make checkMerkleRoot also check the witnessCommit when the block has witnesses.

or make a new function called checkTransactionTrees or something, so that people don't need to worry much about whether the block is witness or not.

Reply:
``` js
    if (!this.transactions) throw new TypeError('Missing transactions')
```

?

Reply:
As for `if (!this.isWitnessBlock()) return false`,  again,  I don't know.
Ideally we'd not have this situation of ambiguity.

Should we "lift" the type somehow?

Reply:
`Transaction` has `hasWitnesses`.

Maybe `hasWitnessCommit`,  force user to check that,  throwing if `checkWitnessCommit` is called and `hasWitnessCommit` is false...

What is the usecase? Merely validation?

Reply:
We do https://github.com/bitcoinjs/bitcoinjs-lib/blob/293116b20ff63473e760039793e5432ffb4a96f2/src/block.js#L164 :-1:

Reply:
Thinking of Lightning Network.

We want to eventually support client side block filtering, which means we only download blocks which match the filter, but we download the full block...

So if we download the full block, and LN uses segwit, we would like a way to verify the witness root as well.

So yeah, merely validation.